blog/content/post/2020-04-03-trenchboot-nlnet-lz-validation.md

```diff
@@ -0,0 +1,887 @@
+Don't worry, I won't give a lecture or bring unfamiliar concepts. I want to show
+you differences between platform without DRTM and with DRTM. It is presented in
+graphical form, so it will be easier to visualize it.
+
+A normal boot process without DRTM enabled, typically looks like this:
+
+![boot without drtm](../../static/img/boot-non-drtm.png)
+
+GRUB bootloader **directly** loads Linux kernel and whereby NixOS boots. As you
+know, by modifying GRUB, Linux kernel and adding special boot modules, we
+enable DRTM. Then, the boot process looks like this:
+
+![boot drtm](../../static/img/boot-drtm.png)
+
+You can clearly see 3 additional elements, which are **slaunch module**,
+**SKINIT** and **Landing Zone**. Their functionality should be already known by
+you. Briefly, it is DRTM stage, when platform boots.
+
+Now, when you can visualize differences, you will learn how to switch your
+platform from first case to second one. Let's do it!
+
+## System customization - enabling DRTM
+
+It is the first thing we need to do. Clean NixOS doesn't meet all requirements.
+First of all, we want to replace default `nixpkgs` with our custom one. Second
+of all, as it doesn't have all necessary packages installed by default, we want
+to add them.
+
+Fortunately, customization of NixOS will demand its **configuration update, but
+not entire system re-installation**. Moreover, there is no need to install every
+single package manually. Of course, it still can be done if you wish. However,
+entire process of enabling DRTM is automated by us to minimize user's effort. So
+much for the introduction - now we can finally run the procedure. Boot to NixOS
+and follow these steps:
+
+1. Install `cachix`
+
+`cachix` is binary cache hosting. It allows to store binary files, so there is
+no need to build them on your own. If it is not very useful for small builds, it
+is very handy for large ones e.g. Linux kernel binary.
+
+    ```bash
+    $ nix-env -iA cachix -f https://cachix.org/api/v1/install
+    ```
+
+2. Add 3mdeb cachix hosting as default.
+
+    ```bash
+    $ cachix use 3mdeb
+    Cachix configuration written to /etc/nixos/cachix.nix.
+    Binary cache 3mdeb configuration written to /etc/nixos/cachix/3mdeb.nix.
+
+    To start using cachix add the following to your /etc/nixos/configuration.nix:
+
+        imports = [ ./cachix.nix ];
+
+    Then run:
+
+        $ sudo nixos-rebuild switch
+    ```
+
+3. Meet above requirement by editing `/etc/nixos/configuration.nix`.
+
+>Probably vim editor is not available at this stage. Instead of vim, you can use
+nano.
+
+    ```bash
+    $ nano /etc/nixos/configuration.nix
+    (...)
+    imports =
+    [ # Include the results of the hardware scan.
+      ./hardware-configuration.nix
+      ./cachix.nix
+    ];
+    (...)
+    ```
+
+>Don't rebuild NixOS yet. It will be done later.
+
+4. Install git package.
+
+    ```bash
+    $ nix-env -iA nixos.git
+    ```
+
+5. Clone
```

Comment:
I'm not sure how this will render, @artur-rs ? Perhaps it would be safer to keep this in one line, even if it means going over 80 characters.

Reply:
Changed to one-line text